src/main/src/main/java/org/geoserver/filters/XFrameOptionsFilter.java

```diff
@@ -50,7 +52,16 @@
  * This header will only be added to HTTPS requests. Default is false.<br>
  * - geoserver.hsts.policy: controls the value of the Strict-Transport-Security header. Default is
  * "max-age=31536000 ; includeSubDomains". Valid options can change the max-age to the desired age
- * in seconds and can omit the includeSubDomains directive.
+ * in seconds and can omit the includeSubDomains directive.<br>
+ * <br>
+ * Two properties control the Content-Security-Policy header to prevent cross-site scripting
+ * attacks:<br>
+ * - geoserver.csp.shouldSetPolicy: controls whether to set the Content-Security-Policy header.
+ * Default is true.<br>
+ * - geoserver.csp.policy: controls the value of the Content-Security-Policy header. Default is to
+ * use the configuration file that will set the header based on the request and server environment
+ * but this property allows the administrator to set a fixed header value, which can be useful in
+ * cases where JavaScript is not needed at all.
  */
 public class XFrameOptionsFilter implements Filter {
```

Comment:
This class started as a way to add XFrameOptions, but over the last few months you pushed more and more headers into it. Can you think of a more appropriate name, given its current function?

Reply:
I considered renaming it to SecurityHeadersFilter but that would be a pretty significant change and would complicate backporting if the renaming wasn't backported.